src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -133,4 +137,20 @@ public static PdaoLoadStatistics getIngestStatistics(FlightContext context) {
     FlightMap workingMap = context.getWorkingMap();
     return workingMap.get(IngestMapKeys.INGEST_STATISTICS, PdaoLoadStatistics.class);
   }
+
+  public static Predicate<FlightContext> noFilesToIngestPredicate() {
+    return flightContext -> {
+      if (flightContext.getFlightClassName().equals(DatasetIngestFlight.class.getName())
+          && Optional.ofNullable(
+                  flightContext.getWorkingMap().get(IngestMapKeys.BULK_LOAD_FILE_MODELS, Set.class))
```

Comment:
Even if we know the flight is `DatasetIngestFlight`, we don't know for sure if `BULK_LOAD_FILE_MODELS` is set or not?

Reply:
If its not set, it'll return `null`, which is why its wrapped in an `Optional.ofNullable`